mono/metadata/unity-liveness.c

```diff
@@ -386,6 +576,63 @@ static void mono_traverse_array (MonoArray* array, LivenessState* state)
 }
 
 
+static void mono_validate_array (MonoArray* array, LivenessState* state)
+{
+	size_t i = 0;
+	gboolean has_references;
+	MonoObject* object = (MonoObject*)array;
+	MonoClass* element_class;
+	size_t elementClassSize;
+	size_t array_length;
+
+	g_assert (object);
+
+
+
+	element_class = GET_VTABLE (object)->klass->element_class;
+	has_references = !mono_class_is_valuetype (element_class);
+	g_assert (element_class->size_inited != 0);
+
+	for (i = 0; i < mono_class_get_field_count (element_class); i++)
```

Comment:
If `has_references` is true we can avoid this loop. We can also break out of it early.

Reply:
Fixed in latest. Added a check to the for loop which should satisfy both situations.